middle_end/flambda2/to_cmm/to_cmm_primitive.ml

```diff
@@ -26,6 +26,25 @@ module C = struct
   include To_cmm_shared
 end
 
+let sign_extend_63 dbg e =
+  let open Cmm in
+  Cop
+    ( Casr,
+      [Cop (Clsl, [e; Cconst_int (1, dbg)], dbg); Cconst_int (1, dbg)],
+      dbg )
+
+let zero_extend_63 dbg e =
+  let open Cmm in
+  Cop (Cand, [e; C.natint_const_untagged dbg 0xFFFFFFFFn(*XXX*)], dbg)
+
+let[@ocaml.warning "-4"] rec low_63 dbg e =
```

Comment:
these functions should move to `cmm_helpers`, and importantly, the tagging helpers should apply `low_63` to the expression to be tagged.

Reply:
Moved, but I am not sure what you mean by "tagging helpers".

Reply:
Also why does `low_63` not follow the code for `low_32` -- the `Cand` case is missing?

Reply:
I thought it was less likely to be useful,
but I certainly see the value of keeping
the functions as similar as possible.

Reply:
I've added the missing `Cand` case.  I also didn't see the functions having been moved to `Cmm_helpers` so I'm doing that too.

Reply:
(Sorry for that, either I did not push it before solving
the conflicts, or the move was accidentally undone
while solving the conflicts...)